### PR TITLE
[semver:minor] HEAT-41: add job for npm outdated

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/HEAT-41

### CODE REVIEW

File: src/jobs/gradle_owasp_dependency_check.yml
Comment (neilmendum) [inline src/jobs/gradle_owasp_dependency_check.yml:60]: Decided to create `slack_failed.tpl` rather than use the existing `slack_failed_with_results.tpl` because that one has a second `context` section which prints _No output, view job for details._ if it cannot parse the output. Wasn't sure about changing the Slack message for the existing jobs, but happy to if folks think don't think it's a big issue

File: src/jobs/npm_outdated.yml
Comment (sldblog) [inline src/jobs/npm_outdated.yml:25]: Do we need installed dependencies (`npm ci`) to run `npm outdated`? Might save a couple of seconds per run.

Comment (sldblog) [reply src/jobs/npm_outdated.yml:25]: Ah, yes. It's reported as `MISSING` if it's not installed. Ignore me

**Review by sldblog — APPROVED** [review]
Looks good to me! (I'm not a maintainer, so maybe needs another review too?)

**Pull request merged**
